File: normalizePoint.js
// Pass point as [lon, lat] array.
function normalizePoint(point) {
    var lon = point[0];
    var lat = point[1];
    
    if (Math.abs(lon) >= 180) {
        while (lon >= 180) {
            lon -= 360;
        }
        while (lon <= -180) {
            lon += 360;
        }
    }

    if (Math.abs(lat) >= 90) {
        while (lat >= 90) {
            lat -= 180;
        }
        while (lat <= -90) {
            lat += 180;
        }
        if (lat !== 0) {
            lat *= -1;
        }
    }
    
    return [lon, lat];
}

File: normalizePoint.py
def normalizePoint(point):
	'''Pass point as a (lon, lat) tuple.'''
	lon, lat = point[0], point[1]

	if abs(lon) >= 180:
		while lon >= 180:
			lon -= 360
		while lon <= -180:
			lon += 360

	if abs(lat) >= 90:
		while lat >= 90:
			lat -= 180
		while lat <= -90:
			lat += 180
		lat *= -1

	return (lon, lat)
